### PR TITLE
Fix exception for non-cs/vb document

### PR DESCRIPTION
**Pull request by genlu**

### CODE REVIEW

File: src/Features/LanguageServer/Protocol/Features/CodeFixes/CodeFixService.cs [hunk 1]
```diff
@@ -255,20 +255,25 @@ internal partial class CodeFixService : ICodeFixService
             CodeActionRequestPriority? priority,
             CancellationToken cancellationToken)
         {
-            if (!(priority is null or CodeActionRequestPriority.Low)
-                || document is not Document sourceDocument)
-            {
+            if (!(priority is null or CodeActionRequestPriority.Low) || document is not Document sourceDocument)
+                return [];
```
Comment (CyrusNajmabadi) [inline src/Features/LanguageServer/Protocol/Features/CodeFixes/CodeFixService.cs:259]: break into multiple checks.

File: src/Features/LanguageServer/Protocol/Features/CodeFixes/CodeFixService.cs [hunk 2]
```diff
@@ -255,20 +255,25 @@ internal partial class CodeFixService : ICodeFixService
...
                 return [];
-            }
 
-            // Expand the fixable range for Copilot diagnostics to containing method.
-            // TODO: Share the below code with other places we compute containing method for Copilot analysis.
-            var root = await sourceDocument.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            var syntaxFacts = sourceDocument.GetRequiredLanguageService<ISyntaxFactsService>();
-            var containingMethod = syntaxFacts.GetContainingMethodDeclaration(root, range.Start, useFullSpan: false);
-            range = containingMethod?.Span ?? range;
+            if (sourceDocument.SupportsSyntaxTree)
```
Comment (CyrusNajmabadi) [inline src/Features/LanguageServer/Protocol/Features/CodeFixes/CodeFixService.cs:265]: i'd just move this above, and bail out immediately.

Comment (genlu) [reply src/Features/LanguageServer/Protocol/Features/CodeFixes/CodeFixService.cs:265]: this is in case we want to support non cs/vb in roslyn workspace

Comment (CyrusNajmabadi) [reply src/Features/LanguageServer/Protocol/Features/CodeFixes/CodeFixService.cs:265]: so, i'm wary about thsi currently as i don't understand the semantics.  specifically, what is hte impact of us passing in the full range of hte document into the methods you're calling.

Also, CodeFixService is firmly lang nutral.  Any lang specific checks (like getting syntax facts, or asking about things like "the containzing method") need to move into a lang-specific api.

That way a langauge like xaml could do the same sort of thing.  they need to be able to say "here's the span i want to limit myself to looking at".

Comment (genlu) [reply src/Features/LanguageServer/Protocol/Features/CodeFixes/CodeFixService.cs:268]: Make sense. How about we get this fix in to unblock insertion. I will follow up with a proper fix next.

File: src/Tools/ExternalAccess/Copilot/Internal/Analyzer/AbstractCopilotCodeAnalysisService.cs [hunk 3]
```diff
@@ -144,7 +145,33 @@ public async Task<ImmutableArray<Diagnostic>> GetCachedDocumentDiagnosticsAsync(
...
+                if (containingMethod?.Span.IntersectsWith(span.Value) is true)
+                    filteredDiagnostics.Add(diagnostic);
+            }
+        }
+        else
+        {
+            foreach (var diagnostic in diagnostics)
+            {
+                if (diagnostic.Location.SourceSpan.IntersectsWith(span.Value))
+                    filteredDiagnostics.Add(diagnostic);
```
Comment (CyrusNajmabadi) [inline src/Tools/ExternalAccess/Copilot/Internal/Analyzer/AbstractCopilotCodeAnalysisService.cs:170]: fine for now.  but we really need this to be done through an ILangService so tha tthe *language* decides what span to be using instead.  right now this is really just not the right way to encode this sort of concept into a feature.

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Tools/ExternalAccess/Copilot/Internal/Analyzer/AbstractCopilotCodeAnalysisService.cs (hunk 3 above)
Comment (genlu) [reply src/Tools/ExternalAccess/Copilot/Internal/Analyzer/AbstractCopilotCodeAnalysisService.cs]: Fixed

**Pull request merged**